devel/ncurses/Portfile

```diff
@@ -44,7 +44,11 @@ configure.args  --enable-widec \
                 --with-pkg-config-libdir="${prefix}/lib/pkgconfig" \
                 --disable-mixed-case
 
-if {[string match *clang* ${configure.cxx}]} {
+platform linux {
+    configure.args-delete --disable-mixed-case
+}
+
+if {${os.platform} eq "darwin" && [string match *clang* ${configure.cxx}]} {
```

Comment:
This idiom is used in tons of ports so if it needs to be changed here it needs to be changed in lots of places. However, why would it need to be changed? Why wouldn't we want to specify what C++ standard library to use on non-macOS systems? If the problem is that MacPorts base isn't setting `configure.cxx_stdlib` (or not setting it correctly) on non-macOS systems, let's fix that in base instead.

Reply:
> If the problem is that MacPorts base isn't setting configure.cxx_stdlib (or not setting it correctly) on non-macOS systems, let's fix that in base instead.

The issue was the following: ``clang: error: invalid library name in argument '-stdlib='``.

You're absolutely right that MacPorts base just isn't setting `configure.cxx_stdlib` correctly. As a result, the build completes if I set `cxx_stdlib libstdc++` in `macports.conf`. I should be able to send a PR to fix this at some point.